### PR TITLE
ESP32-S3 initial commit

### PR DESCRIPTION
**Pull request by Sour-Codes**
## Description

Initial development of ESP32-S3 support.

## Motivation and Context

- To provide support to ESP32-S3.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: targets/ESP32/_common/ESP32_S3_DeviceMapping.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+//  "    ADC2 channels 10 - 19
+// TODO review ADC channels for ESP32_S3
+int8_t Esp32_ADC_DevicePinMap[6] = {
+    // 0   1   2   3   4   5   
+    0, 1, 2, 3, 4, 5};
+
+//  I2S
+//  1 device I2S1
+//  Map pins various pins. If not used, I2S_PIN_NO_CHANGE is used
+int8_t Esp32_I2S_DevicePinMap[I2S_NUM_MAX][5] = {
```
Comment (Ellerbach) [inline targets/ESP32/_common/ESP32_S3_DeviceMapping.cpp:83]: it has actually 2 I2S devices: https://docs.espressif.com/projects/esp-idf/en/v4.4.4/esp32s3/api-reference/peripherals/i2s.html?highlight=i2s

From the doc: ESP32-S3 contains two I2S peripheral(s). These peripherals can be configured to input and output sample data via the I2S driver.

File: targets/ESP32/_common/ESP32_S3_DeviceMapping.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+//  Map pins Data & Clock
+int8_t Esp32_I2C_DevicePinMap[I2C_NUM_MAX][2] = {
+    // I2C1 - pins 18, 19,
+    {I2C1_DATA, I2C1_CLOCK}};
+
+// =============================================
+//  LED PWM
+//  16 channels LED1 to LED16  or PWM1 to PWM16
+//  Map pins Data & Clock
+int8_t Esp32_LED_DevicePinMap[16] = {
```
Comment (Ellerbach) [inline targets/ESP32/_common/ESP32_S3_DeviceMapping.cpp:49]: not 100% sure for this one as it seems to have 8 channels only: https://docs.espressif.com/projects/esp-idf/en/v4.4.4/esp32s3/api-reference/peripherals/ledc.html?highlight=led%20pwm#_CPPv414ledc_channel_t

**Review by Ellerbach — COMMENTED** [review]
Huge thanks! That looks good to me all up. But I'm sure @josesimoes will do a proper deeper review!

In the mean time, some elements to adjust regarding the features available for the S3

**Comment by Sour-Codes** [comment]
I honestly have no idea what I'm doing here 😅 - Just trying to do my part.

Please, if it's not too much to ask, how exactly were you alerted to the changes to [ADC](https://docs.espressif.com/projects/esp-idf/en/v4.4.4/esp32s3/api-reference/peripherals/adc.html?highlight=adc) as well as the changes to [I2S](https://docs.espressif.com/projects/esp-idf/en/v4.4.4/esp32s3/api-reference/peripherals/i2s.html?highlight=i2s)?

Seems like I need to bookmark and do a deep dive in Espressif's API reference.

I would like to be well versed to be able to provide support to the ESP32 cores as they are released for the community.

**Pull request closed without merging**